app/models/plan.rb

```diff
@@ -2,11 +2,25 @@ class Plan < ApplicationRecord
   before_create :generate_token
   validates :name, :price, presence: true
   belongs_to :customer
+  has_many :plan_price_history, dependent: :destroy
+  after_save :save_price_history
```

Comment:
```suggestion
  after_commit :save_price_history
```

Aqui talvez fosse melhor usar `after_commit` pra garantir que o plano foi salvo no banco antes de criar o histórico, mas vocês podem testar.